contracts/src/v0.8/automation/upkeeps/LinkAvailableBalanceMonitor.sol

```diff
@@ -135,20 +119,23 @@ contract LinkAvailableBalanceMonitor is ConfirmedOwner, Pausable, AutomationComp
   function sampleUnderfundedAddresses() public view returns (address[] memory) {
     uint16 maxPerform = s_maxPerform;
     uint16 maxCheck = s_maxCheck;
-    uint256 numTargets = s_watchList.length();
-    uint256 idx = uint256(blockhash(block.number - 1)) % numTargets; // start at random index, to distribute load
+    uint256 numTargets = s_watchList.length;
+    uint256 idx = uint256(blockhash(block.number - 1)) % numTargets;
     uint256 numToCheck = numTargets < maxCheck ? numTargets : maxCheck;
     uint256 numFound = 0;
+    uint256 minWaitPeriod = s_minWaitPeriodSeconds;
     address[] memory targetsToFund = new address[](maxPerform);
+    MonitoredAddress memory target;
     for (
       uint256 numChecked = 0;
       numChecked < numToCheck;
       (idx, numChecked) = ((idx + 1) % numTargets, numChecked + 1)
     ) {
-      (address target, uint256 minBalance) = s_watchList.at(idx);
-      (bool needsFunding, ) = _needsFunding(target, minBalance);
-      if (needsFunding) {
-        targetsToFund[numFound] = target;
+      address targetAddress = s_watchList[idx];
+      target = s_targets[targetAddress];
+      (bool needsFunding, ) = _needsFunding(targetAddress, target.minBalance);
+      if (needsFunding && target.lastTopUpTimestamp + minWaitPeriod <= block.timestamp) {
```

Comment:
Would it make sense to move the `target.lastTopUpTimestamp + minWaitPeriod <= block.timestamp` into `_needsFunding()`?

Reply:
Yeah, agree, that way we don't overcheck it. Moving it.